### PR TITLE
Expand implementation of lax_numpy._quantile to allow the input of a …

### LINKED ISSUE #9277: jax.numpy.nanpercentile with axis as tuple
currently, the following code results in a NotImplementedError:
```
jax.numpy.nanpercentile(jnp.ones((3,3,12,24)), 99.9, axis=(0,1,2))
```

```
 File "/home/clemens/.local/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py", line 6359, in _quantile                                       
    raise NotImplementedError("Tuple values for axis are not implemented")   
```
However the docstring actually explicitly mentions tuple as a valid option for axis: https://jax.readthedocs.io/en/latest/_autosummary/jax.numpy.nanpercentile.html

same for jax.numpy.percentile.html

### PR DESCRIPTION
**Pull request by MichaelMarien**
Expand implementation of lax_numpy._quantile to allow the input of a tuple as axis argument. Fixes #9277 

- introduced and changed corresponding numpy code to swap and reshape axis to be quantiled
- introduced code to accomodate the reintroduction of those axes if keepdims=True
- added testcases

### CODE REVIEW

File: jax/_src/numpy/lax_numpy.py [hunk 1]
```diff
@@ -6406,13 +6406,29 @@ def _quantile(a, q, axis, interpolation, keepdims, squash_nans):
...
   elif isinstance(axis, tuple):
-    raise NotImplementedError("Tuple values for axis are not implemented")
+    keepdim = list(shape(a))
+    nd = ndim(a)
+    axis = tuple([_canonicalize_axis(ax, nd) for ax in axis])
+
+    for ax in axis:
+      keepdim[ax] = 1
+
+    keep = set(range(nd)) - set(axis)
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:6423]: Numpy errors on repeated axes. Perhaps JAX should do this too?

File: jax/_src/numpy/lax_numpy.py [hunk 2]
```diff
@@ -6406,13 +6406,29 @@ def _quantile(a, q, axis, interpolation, keepdims, squash_nans):
...
+    nkeep = len(keep)
+    # swap axis that should not be reduced to front
+    for i, s in enumerate(sorted(keep)):
+      a = a.swapaxes(i, s)
+    # merge reduced axis
+    a = reshape(a, shape(a)[:nkeep] + (-1,))
```
Comment (jakevdp) [inline jax/_src/numpy/lax_numpy.py:6429]: swapping axes and reshaping can be somewhat expensive. I think `lax.reshape` is general enough to do this axis swapping plus the reshape in a single XLA call: we should use that here.

**Review by jakevdp — CHANGES_REQUESTED** [review]
Looks great, thanks for taking this on! Just one substantial comment below, regarding a more efficient implementation.

File: jax/_src/numpy/lax_numpy.py (hunk 1 above)
Comment (MichaelMarien) [reply jax/_src/numpy/lax_numpy.py:6423]: Agreed!

File: jax/_src/numpy/lax_numpy.py (hunk 2 above)
Comment (MichaelMarien) [reply jax/_src/numpy/lax_numpy.py:6429]: Totally, did not know about the existence of the `dimensions` arg in `lax.reshape`.

**Comment by MichaelMarien** [comment]
@jakevdp tried to address your comments, let me know what you think and thanks for the review effort.

**Review by jakevdp — APPROVED** [review]
Looks awesome, thanks for all the quick responses on the feedback!

Last thing: could you please squash all the changes into a single commit (see https://jax.readthedocs.io/en/latest/contributing.html#single-change-commits-and-pull-requests). Once you do, we can start the merge process

**Comment by MichaelMarien** [comment]
> Last thing: could you please squash all the changes into a single commit (see https://jax.readthedocs.io/en/latest/contributing.html#single-change-commits-and-pull-requests). Once you do, we can start the merge process

Made a bit of a git mess at first, but I think it's fine now.

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
Thanks for the contribution!
